packages/editor/src/lib/editor/Editor.ts

```diff
@@ -4819,42 +4826,40 @@ export class Editor extends EventEmitter<TLEventMap> {
 
 	/* -------------------- Bindings -------------------- */
 
+	@computed
+	private _getBindingsIndexCache() {
+		const index = bindingsIndex(this)
+		return this.store.createComputedCache<TLBinding[], TLShape>('bindingsIndex', (shape) => {
+			return index.get()[shape.id]
+		})
+	}
+
 	getBinding(id: TLBindingId): TLBinding | undefined {
 		return this.store.get(id) as TLBinding | undefined
 	}
 
-	// TODO(alex) #bindings - cache `allBindings` getters and derive type-specific ones from them
 	getBindingsFromShape<Binding extends TLUnknownBinding = TLBinding>(
 		shape: TLShape | TLShapeId,
-		type: Binding['type']
+		type?: Binding['type']
 	): Binding[] {
 		const id = typeof shape === 'string' ? shape : shape.id
-		return this.store.query.exec('binding', {
-			fromId: { eq: id },
-			type: { eq: type },
-		}) as Binding[]
+		return this.getBindingsInvolvingShape(id, type).filter((b) => b.fromId === id) as Binding[]
```

Comment:
this seems like it no longer does the `type` check. can you restore?

also, i think i would make type non-optional on this api. as these APIs are directional, i think they're fairly binding-type specific and more general binding queries are better answered by `getBindingsInvolvingShape`. I imagine that these two APIs are mostly going to be used by binding implementors for their own bindings

Reply:
the type check happens in the underlying getBindingsInvolvingShape. I'll make it non-optional too, good call.

Reply:
oh i missed that! thx

Reply:
done